Modernize the project followup

0xced opened the pull request:
Since #55 became unmergeable, I reapplied some of the fixes in this new pull request.

* In the sample app, target `net6.0-windows` and `net8.0-windows` instead of suppressing `CA1416`
* Declare [SupportedOSPlatform("windows")] at the assembly level, ~with the help of the [Polyfill][1] package to avoid conditional compilation~
* Always return the NullSink when not running on Windows no matter what platform the package is compiled for
* Simplify the conditional package reference to `System.Diagnostics.EventLog` with `$(TargetFrameworkIdentifier) != '.NETFramework'`
* Remove the `Nullable` package (the ~`Polyfill`~ `PolySharp` package has everything needed for nullable annotations on .NET Standard 2.0 and more)
* Cleanup some project properties
  * `GenerateDocumentationFile` and `TreatWarningsAsErrors` were duplicated
  * `PackageTags` was duplicated (and wrong)
  * Fix `RepositoryUrl` (serilog-sinks-eventlog instead of serilog-sinks-async)
  * Remove `RepositoryType` = `git` (which is the default)

[1]: https://github.com/SimonCropp/Polyfill

0xced committed:
Target net8.0-windows instead of suppressing CA1416 in sample app

0xced committed:
Simplify everything around conditional compilation

* Declare [SupportedOSPlatform("windows")] at the assembly level, with the help of the [Polyfill][1] package to avoid conditional compilation.
* Always return the NullSink when not running on Windows no matter what platform the package is compiled for.
* Simplify the conditional package reference to System.Diagnostics.EventLog with $(TargetFrameworkIdentifier) != '.NETFramework'
* Remove the Nullable package (the Polyfill package has everything needed for nullable annotations on .NET Standard 2.0 and more)

[1]: https://github.com/SimonCropp/Polyfill

0xced committed:
Cleanup some project properties

* GenerateDocumentationFile and TreatWarningsAsErrors were duplicated
* PackageTags was duplicated (and wrong)
* Fix RepositoryUrl (serilog-sinks-eventlog instead of serilog-sinks-async)
* Remove RepositoryType = git (which is the default)

Numpsy commented:
> Fix RepositoryUrl (serilog-sinks-eventlog instead of serilog-sinks-async

Is there a need to specify the URL in the project file, or can it just be left to SourceLink to get it itself?

Numpsy commented on src/Serilog.Sinks.EventLog/LoggerConfigurationEventLogExtensions.cs:
```diff
@@ -110,12 +103,11 @@ public static class LoggerConfigurationEventLogExtensions
         IEventIdProvider? eventIdProvider = null)
     {
-#if FEATURE_RUNTIMEINFORMATION
         if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```
Should it be using !OperatingSystem.IsWindows() where available? (could possibly be better for trimming on not-windows platforms?)

0xced replied:
Indeed, just addressed in f114631e8329a280f3d00bebba0705e891a05c75.

I did not notice it would not even compile on `net462` because I was on macOS. With this last commit, the build on AppVeyor should pass now.

0xced committed:
Use PublishRepositoryUrl = true

So that the repository URL is always correct

0xced commented:
> > Fix RepositoryUrl (serilog-sinks-eventlog instead of serilog-sinks-async
> 
> Is there a need to specify the URL in the project file, or can it just be left to SourceLink to get it itself?

Absolutely! It just needs `PublishRepositoryUrl` actually. I just did it in 09fbc88c95d16d450c24e354d025b4fa4d1f559a.

0xced committed:
Fix Windows detection

Use OperatingSystem.IsWindows() when available (.NET 5+), use always true for net46x and RuntimeInformation.IsOSPlatform() otherwise (.NET Standard and .NET Framework != 4.6)

0xced committed:
Use PolySharp instead of Polyfill

PolySharp includes much less polyfills than Polyfill, thus reducing the final package size.

0xced committed:
Fix compilation

NET46X was defined by the PolyFill package, NET462 is defined by the .NET SDK

0xced commented:
I finally switched from [Polyfill][1] to [PolySharp][2] to reduce the package size since PolySharp does not include unused polyfills. The final NuGet package size was thus reduced from [299KB][3] to [84KB][4]

[1]: https://github.com/SimonCropp/Polyfill
[2]: https://github.com/Sergio0694/PolySharp
[3]: https://ci.appveyor.com/project/serilog/serilog-sinks-eventlog/builds/50076908/artifacts
[4]: https://ci.appveyor.com/project/serilog/serilog-sinks-eventlog/builds/50076934/artifacts

nblumhardt commented:
LGTM! Thanks @0xced :sunglasses:
